File: Code/Frontend/src/app/account/account.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {Observable} from 'rxjs';

import {Customer, Order, Product} from '../shared/entity/models';
import {WSApi} from '../shared/entity/constants';

@Injectable({
    providedIn: 'root'
})
export class AccountService {

    constructor(private http: HttpClient) {
    }

    updateProfile(customer: Customer): Observable<Customer> {
        const URL = `/${WSApi.BASE_URL}/${WSApi.CUSTOMER}/${WSApi.PROFILE}/${WSApi.UPDATE}`;
        return this.http.post<Customer>(URL, customer);
    }

    getOrders(customerId: number): Observable<Order[]> {
        const URL = `/${WSApi.BASE_URL}/${WSApi.CUSTOMER}/${customerId}/${WSApi.ORDERS}`;
        return this.http.get<Order[]>(URL);
    }

    modifyProduct(product: Product, customerId: number, type: string, removeProduct: number): Observable<void> {
        const URL = `/${WSApi.BASE_URL}/${WSApi.CUSTOMER}/${customerId}/${WSApi.PRODUCT}/${type}?removeProduct=${removeProduct}`;
        return this.http.post<void>(URL, product);
    }

    clearProducts(customerId: number, type: string): Observable<void> {
        const URL = `/${WSApi.BASE_URL}/${WSApi.CUSTOMER}/${customerId}/${WSApi.CLEAR}/${type}`;
        return this.http.post<void>(URL, {});
    }

    checkOut(cartProducts: Product[], customerId: number): Observable<void> {
        const URL = `/${WSApi.BASE_URL}/${WSApi.CUSTOMER}/${customerId}/${WSApi.CHECKOUT}`;
        return this.http.post<void>(URL, cartProducts);
    }

}

File: Code/Backend/src/main/java/com/webstore/webStore/service/account/AccountService.java
package com.webstore.webStore.service.account;

import com.webstore.webStore.entity.Customer;
import com.webstore.webStore.entity.Order;
import com.webstore.webStore.entity.Product;
import com.webstore.webStore.entity.WishlistCart;
import org.springframework.stereotype.Service;

import java.util.List;

@Service("accountService")
public interface AccountService {

    // Update Customer Profile
    Customer updateProfile(Customer customer);

    // Get Customer Orders
    List<Order> getOrders(Integer customerID);

    // Get Products In Wishlist/Cart
    List<WishlistCart> getProducts(Integer customerID, String type);

    // Add/Remove Product In Wishlist/Cart
    void modifyProduct(Product product, Integer customerID, String type, Integer removeProduct);

    // Clear Products In Wishlist/Cart
    void clearProducts(Integer customerID, String type);

    // Cart Checkout
    void checkOut(List<Product> cartProducts, Integer customerID);
}
